src/utils/config.js

```diff
@@ -50,63 +50,43 @@ const defaultJanusServer = () => {
 const replacePlaceholders = (text) => text.replace('%{hostname}', window.location.hostname);
 
 /**
- * Parses a piece of JSON returning an empty object if it is not valid.
+ * Parses and builds the configuration object.
  *
  * @param {string} text Configuration (in JSON format).
  * @returns {Object}
  */
-const parseJSON = (text) => {
+function parseConfig(text) {
+  const replacedText = replacePlaceholders(text);
+  let config;
   try {
-    return JSON.parse(text);
+    config = JSON.parse(replacedText);
   } catch (e) {
-    console.warn('The configuration is not valid JSON.', e);
-    return {};
+    console.error(`Could not parse the configuration file: ${e}. Using default values.`);
+    return defaultConfig();
   }
-};
 
-/**
- * Parses and buids the configuration object.
- *
- * @param {string} text Configuration (in JSON format).
- * @returns {Object}
- */
-function parseConfig(text) {
-  const replacedText = replacePlaceholders(text);
-  const config = parseJSON(replacedText);
-  Object.keys(config).forEach((k) => {
+  Object.keys(config).forEach(k => {
     if (config[k] === undefined || config[k] === null) {
       delete config[k];
     }
   });
   return { ...defaultConfig(), ...config };
 }
 
-const READY_STATE_DONE = 4;
-const STATUS_SUCCESS = 200;
-
 /**
- * Fetches and parses the configuration from the `config.json` file.
+ * Retrieves the configuration
  *
- * @returns {Object}
+ * It tries to get a custom configuration from `/config.json`. If a valid configuration
```

Comment:
For other readers: example available in https://github.com/jangouts/jangouts/blob/215b1b107d332ea0108869213755dd8573cd7c98/public/config.json.sample